### DISCUSSION TITLE
Tailwind is one feature away from perfection

### DISCUSSION THREAD

**Discussion by danirdd92** [tailwindlabs/tailwindcss#10095] [Ideas]
**I first want to say I'm in love with the work you guys have done, you made me fall in love with CSS as a front-end developer.** 

However, there is one feature IMO that is missing to make tailwind the perfect CSS engine of choice, lets call it "Nested Selectors".

Let me explain, if you look at this example: 

```
"flex h-full flex-col md:flex-row md:items-center [&_li]:ml-6 [&_li]:border-b [&_li]:border-grey-dark md:[&_li]:border-none
ease-in [&_a]:flex [&_a]:h-navigation-height [&_a]:w-full [&_a]:translate-y-8 [&_a]:items-center [&_a]:text-lg [&_a]:transition-[color,transform] [&_a]:duration-300 md:[&_a]:translate-y-0 md:[&_a]:text-sm [&_a]:md:transition-colors [&_a:hover]:text-grey
"}
            >
```
 
[&_a] is written 11 times in order to account for all the styles in a nested <a> tag. And the fact it's being displayed as a string in the ide and not highlighted differently makes looking at these styles to derive meaning very cumbersome (please do tell me if there's an extension for that).

IMO, it breaks the whole appeal of tailwind, that is looking at the text and understand the design's intention just by looking at the code. And I'm not even mentioning writing this over and over repeatedly.

My suggestion is pretty obvious, nest all selector related style under a single selector, this should be able to be done recursively 
if (e.g., there can be many [&_li] classes inside md:{}]
```
'flex h-full flex-col ease-in
md:{flex-row items-center [&_li]:{border-none text-sm transition-colors translate-y-0}}
[&_li]:{ml-6 border-b border-grey-dark} 
[&_a]:{flex h-navigation-height w-full translate-y-8 items-center text-lg transition-[color,transform] duration-300 hover:text-grey}'
```

I think we can all agree the classes are way more readable after being nested inside a selector, also there's the added benefit that a formatter can indent the code in a more human readable way like I've new lined each selector here.

**Comment by wongjn** [comment]
[Here's a response to a similar suggestion](https://github.com/tailwindlabs/tailwindcss/discussions/9053#discussioncomment-3350470):

> We get this question so many times, so I would like to redirect you here for more information: https://twitter.com/adamwathan/status/1461519820411789314

**Reply by danirdd92** [reply]
I've commented on their thread. I hope my suggestion is worth anything

**Comment by Noriller** [comment]
Hey!

I'm actually taking a stab at doing that: https://www.npmjs.com/package/easy-tailwind

You can play with it here: https://stackblitz.com/edit/easy-tailwind?file=src/App.jsx

I'm still polishing the docs (and finding some problems here and there), but I think it's already at a place you can play around and I hope to finish the last touches to release it this weekend at worst.

It turns this:

```js
<div
  className={e(
    'text-lg font-medium text-black',
    {
      hover: 'underline decoration-black',
      sm: [
        'text-base text-blue-500',
        {
          hover: 'decoration-cyan-500',
        },
      ],
      lg: [
        'text-2xl text-green-500',
        {
          hover: 'decoration-amber-500',
        },
      ],
    }
  )}
>
  EasyTailwind!!!
</div>
```

And different than: https://twitter.com/adamwathan/status/1461519820411789314
"All" it does is to turn what you write into this:

```js
"text-lg font-medium text-black hover:underline hover:decoration-black sm:text-base sm:text-blue-500 sm:hover:decoration-cyan-500 lg:text-2xl lg:text-green-500 lg:hover:decoration-amber-500"
```
